ARM Support

The pre-compiled macOS binary appears to be x86 only. Would it be possible to have a universal build supporting ARM architecture?
When building from source using `cargo install ouch` rustup first installs an x86-64 toolchain (in addition to the arm64 toolchain) on my computer and later fails during the compilation:

```
   Compiling ouch v0.1.5
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "-L" "~/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/var/folders/_g/nfj48t2948s35nnrcs4p3q_c0000gn/T/cargo-install1namSU/release/deps/ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o" "-o" "/var/folders/_g/nfj48t2948s35nnrcs4p3q_c0000gn/T/cargo-install1namSU/release/deps/ouch-c1fc9f0cbd401a2e" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/var/folders/_g/nfj48t2948s35nnrcs4p3q_c0000gn/T/cargo-install1namSU/release/deps" "-L" "/var/folders/_g/nfj48t2948s35nnrcs4p3q_c0000gn/T/cargo-install1namSU/release/build/bzip2-sys-e23d54bac5dc72d9/out/lib" "-L" "/opt/homebrew/Cellar/xz/5.2.5/lib" "-L" "/Users/salomon/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/var/folders/_g/nfj48t2948s35nnrcs4p3q_c0000gn/T/rustcnCaz4b/libbzip2_sys-ad219373a4b8ba7e.rlib" "/Users/salomon/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-160a7dd17f2dc334.rlib" "-llzma" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv"
  = note: ld: warning: ignoring file /opt/homebrew/Cellar/xz/5.2.5/lib/liblzma.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
          Undefined symbols for architecture x86_64:
            "_lzma_code", referenced from:
                _$LT$xz2..read..XzDecoder$LT$R$GT$$u20$as$u20$std..io..Read$GT$::read::hb1428dcfb9c41431 in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
                ouch::compressors::lzma::LzmaCompressor::compress_bytes::h94f82ba83697a63a in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
                xz2::write::XzEncoder$LT$W$GT$::try_finish::hcd128a9a358b26ee in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
            "_lzma_easy_encoder", referenced from:
                ouch::compressors::lzma::LzmaCompressor::compress_bytes::h94f82ba83697a63a in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
            "_lzma_auto_decoder", referenced from:
                ouch::decompressors::to_memory::DecompressorToMemory::decompress::hd679a1dec8865489 in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
            "_lzma_end", referenced from:
                ouch::decompressors::to_memory::DecompressorToMemory::decompress::hd679a1dec8865489 in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
                core::ptr::drop_in_place$LT$xz2..read..XzDecoder$LT$alloc..boxed..Box$LT$dyn$u20$std..io..Read$u2b$core..marker..Send$GT$$GT$$GT$::hf7ec9572769d8944 in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
                core::ptr::drop_in_place$LT$xz2..stream..Stream$GT$::h6048c3516c7203ec in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
                ouch::compressors::lzma::LzmaCompressor::compress_bytes::h94f82ba83697a63a in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
                core::ptr::drop_in_place$LT$xz2..write..XzEncoder$LT$alloc..vec..Vec$LT$u8$GT$$GT$$GT$::h5164a69ef0e186df in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
                core::ptr::drop_in_place$LT$xz2..stream..Stream$GT$::h92dfb9102f4d6f54 in ouch-c1fc9f0cbd401a2e.ouch.1mb7mks5-cgu.0.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/.github/workflows/build-artifacts-and-run-tests.yml
+++ b/.github/workflows/build-artifacts-and-run-tests.yml
@@ -48,6 +48,7 @@ jobs:
           - x86_64-pc-windows-msvc
           - aarch64-pc-windows-msvc
           - x86_64-apple-darwin
+          - aarch64-apple-darwin
           # cross
           - x86_64-unknown-linux-musl
           - aarch64-unknown-linux-gnu
@@ -64,6 +65,8 @@ jobs:
           - target: aarch64-pc-windows-msvc
             os: windows-latest
           - target: x86_64-apple-darwin
+            os: macos-13
+          - target: aarch64-apple-darwin
             os: macos-latest
           # targets that use cross
           - target: x86_64-unknown-linux-musl

--- a/.github/workflows/build-artifacts-and-run-tests.yml
+++ b/.github/workflows/build-artifacts-and-run-tests.yml
@@ -65,7 +65,7 @@ jobs:
           - target: aarch64-pc-windows-msvc
             os: windows-latest
           - target: x86_64-apple-darwin
-            os: macos-13
+            os: macos-latest
           - target: aarch64-apple-darwin
             os: macos-latest
           # targets that use cross

